File: app/controllers/HomeController.scala
package controllers

import com.typesafe.config.Config
import example.myapp.helloworld.grpc.{ GreeterServiceClient, HelloReply, HelloRequest }
import javax.inject.Inject
import play.api.mvc._

import scala.concurrent.{ ExecutionContext, Future }

class HomeController @Inject() (greeterServiceClient: GreeterServiceClient, config:Config)(implicit ec: ExecutionContext) extends InjectedController {

  def index = Action.async {
    val request = HelloRequest("Caplin")
    // create a gRPC request
    val reply: Future[HelloReply] = greeterServiceClient.sayHello(request)
    // forward the gRPC response back as a plain String on an HTTP response
    reply.map(_.message).map(m => Ok(m))
  }

}

File: app/controllers/HomeController.java
package controllers;

import example.myapp.helloworld.grpc.GreeterServiceClient;
import example.myapp.helloworld.grpc.HelloReply;
import example.myapp.helloworld.grpc.HelloRequest;
import play.mvc.Controller;
import play.mvc.Result;
import play.mvc.Results;

import javax.inject.Inject;
import java.util.concurrent.CompletionStage;

public class HomeController extends Controller {

    private final GreeterServiceClient greeterServiceClient;

    @Inject
    public HomeController(GreeterServiceClient greeterServiceClient) {
        this.greeterServiceClient = greeterServiceClient;
    }
    public CompletionStage<Result> index() {
        HelloRequest request = HelloRequest.newBuilder().setName("Caplin").build();
        CompletionStage<HelloReply> reply = greeterServiceClient.sayHello(request);

        return reply.thenApply(HelloReply::getMessage).thenApply(Results::ok);
    }

}
